src/v/redpanda/admin_server.cc

```diff
@@ -2929,6 +2930,130 @@ void admin_server::register_transaction_routes() {
+    }
+    auto doc = parse_json_body(*req);
+    std::optional<cluster::diskcheck_opts> dto;
+    std::optional<cluster::netcheck_opts> nto;
+    if (!doc.HasMember("tests")) {
+        dto = cluster::diskcheck_opts::from_json(doc);
+        nto = cluster::netcheck_opts::from_json(doc);
+    } else {
+        for (const auto& name : doc["tests"].GetArray()) {
+            if (name == "disk") {
+                dto = cluster::diskcheck_opts::from_json(doc);
+            } else if (name == "network") {
+                nto = cluster::netcheck_opts::from_json(doc);
+            }
+        }
+    }
+    try {
+        auto tid = co_await _self_test_frontend.invoke_on(
+          cluster::self_test_frontend::shard,
+          [dto, nto](auto& self_test_frontend) {
+              return self_test_frontend.start_test(dto, nto);
+          });
+        vlog(logger.info, "Request to start self test succeeded: {}", tid);
+        co_return ss::json::json_return_type(tid);
+    } catch (const std::exception& ex) {
+        throw ss::httpd::base_exception(
+          fmt::format("Failed to start self test, reason: {}", ex),
+          ss::httpd::reply::status_type::service_unavailable);
+    }
+}
+
+ss::future<ss::json::json_return_type>
+admin_server::self_test_stop_handler(std::unique_ptr<ss::httpd::request> req) {
+    if (need_redirect_to_leader(model::controller_ntp, _metadata_cache)) {
+        vlog(logger.info, "Need to redirect self_test_stop request");
+        throw co_await redirect_to_leader(*req, model::controller_ntp);
+    }
+    auto r = co_await _self_test_frontend.invoke_on(
+      cluster::self_test_frontend::shard,
+      [](auto& self_test_frontend) { return self_test_frontend.stop_test(); });
+    if (!r.finished()) {
+        throw ss::httpd::base_exception(
+          fmt::format(
+            "Failed to stop one or more self_test jobs: {}",
+            r.active_participant_ids()),
+          ss::httpd::reply::status_type::service_unavailable);
+    }
+    vlog(logger.info, "Request to stop self test succeeded");
+    co_return ss::json::json_void();
+}
+
+static ss::httpd::debug_json::self_test_result
+self_test_result_to_json(const cluster::self_test_result& str) {
+    ss::httpd::debug_json::self_test_result r;
+    if (str.error) {
+        r.error = *str.error;
+        return r;
+    }
+    if (str.warning) {
+        r.warning = *str.warning;
+    }
+    r.p50 = str.p50;
+    r.p90 = str.p90;
+    r.p99 = str.p99;
+    r.p999 = str.p999;
+    r.max_latency = str.max;
+    r.rps = str.rps;
+    r.bps = str.bps;
+    r.test_id = ss::sstring(str.test_id);
+    r.name = str.name;
+    r.duration = std::chrono::duration_cast<std::chrono::milliseconds>(
+                   str.duration)
+                   .count();
+    return r;
+}
+
+ss::future<ss::json::json_return_type>
+admin_server::self_test_get_results_handler(
+  std::unique_ptr<ss::httpd::request>) {
+    namespace dbg_ns = ss::httpd::debug_json;
```

Comment:
nit: for posterity I think it's worth adding a comment mentioning/explaining why we're explicitly not redirecting to the leader for this operation (it's read-only and inexpensive to send anywhere). It's otherwise tempting to look at this and assume it was an oversight